cg/cli/workflow/rnafusion/base.py

```diff
@@ -348,8 +327,26 @@ def store_housekeeper(context: CGConfig, case_id: str, dry_run: bool) -> None:
 def store(context: click.Context, case_id: str, dry_run: bool) -> None:
     """Generate deliverables files for a case and store in Housekeeper if they
     pass QC metrics checks."""
-    LOG.info("Generating metrics file and performing QC checks for %s", case_id)
-    context.invoke(metrics_deliver, case_id=case_id, dry_run=dry_run)
+    analysis_api: RnafusionAnalysisAPI = context.obj.meta_apis[MetaApis.ANALYSIS_API]
+
+    if not any(
+        [
+            analysis_api.trailblazer_api.is_latest_analysis_completed(case_id=case_id),
+            analysis_api.trailblazer_api.is_latest_analysis_qc(case_id=case_id),
+        ]
+    ):
+        LOG.error(
+            "Case not stored. Trailblazer status must be either QC or COMPLETE to be able to store."
+        )
+        return
+
+    # Avoid storing a case without QC checks previously performed.
+    if (
+        analysis_api.trailblazer_api.is_latest_analysis_qc(case_id=case_id)
```

Comment:
Since this line is being repeated maybe it's a good idea to define some variables of this conditions on top:

is_latest_analysis_qc: bool = ....